etc/shadow-maint/userdel-pre.d/01-kill_user_procs.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+# Check user exists, and if so, send sigkill to processes that the user owns
+
+RUNNING=`ps -eo user | grep -Fx "$SUBJECT" | wc -l`
+
+# if the user does not exist, RUNNING will be 0
+
+if [ "${RUNNING}x" = "0x" ]; then
+  exit 0
+fi
+
+ps -eo pid,user | grep -E "^\s*[0-9]+\s+${SUBJECT}$" | awk '{ print $1 }' | xargs kill -9
```

Comment:
It's possible that my busybox is just bad, but it doesn't insert a space between the two columns, so only a few of the pids get listed (those which have my username in the cmdline).  Example output:

902725root     [kworker/5:1-mm_]

I wonder whether we're not better off just checking whether killall exists on the system, using killall -u ${SUBJECT} if so, and just leaving things be if it does not exist.